DEVELOPING.md

```diff
@@ -132,6 +132,49 @@ sequenceDiagram
 ```
 
 
+### Syft Catalogers
+
+##### Summary
+Catalogers are the way in which syft is able to identify and construct packages given some amount of source metadata. 
+For example, syft can use a `package.json` file in conjunction with a `package-lock.json` file to build a list of javascript packages based on the joining metadata:
+- [Cataloger Construction](https://github.com/anchore/syft/blob/e3d6ffd30e44428b898675922a0474221a7f7dc7/syft/pkg/cataloger/javascript/cataloger.go#L16-L21)
+- [Package Lock Parsing](https://github.com/anchore/syft/blob/e3d6ffd30e44428b898675922a0474221a7f7dc7/syft/pkg/cataloger/javascript/parse_package_lock.go#L43-L90)
+
+#### Building a new Cataloger
+Catalogers must fulfill the interface found here. This means that when building a new cataloger, the new struct must implement both method signatures of `Catalog` and `Name`:
+- [Cataloger Interface](https://github.com/anchore/syft/blob/main/syft/pkg/cataloger.go)
```

Comment:
Small stylistic suggestion: include the links inline in the text, like:

> Catalogers must [fulfill the interface found here](https://github.com/anchore/syft/blob/main/syft/pkg/cataloger.go)

Reply:
latest commit takes the suggestions and inlines everything